### PR TITLE
UI - unable to obtain info summary table updates

### PR DESCRIPTION
**Pull request by kialj876**
*Issue:*https://github.com/bcgov/entity/issues/21660

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by kialj876** [comment]
still updating tests

File: btr-web/btr-main-app/components/bcros/AddressDisplay.vue [hunk 1]
```diff
@@ -24,11 +24,11 @@ const props = defineProps<{ address: Partial<AddressI> }>()
 
 const addressData = computed((): string[] => {
   return [
-    props.address.line1,
-    props.address.line2,
+    props.address.line1 ?? '',
+    props.address.line2 ?? '',
     [props.address.city, props.address.region, props.address.postalCode].filter(val => !!val).join(' '),
     // NOTE: getName throws an error when called with undefined
-    getName(props.address.country.alpha_2 ?? '') || props.address.country.name
```
Comment (kialj876) [inline btr-web/btr-main-app/components/bcros/AddressDisplay.vue:31]: this was needed to render the row without entering address info

File: btr-web/btr-main-app/components/individual-person/summary-table/Index.vue [hunk 2]
```diff
@@ -193,77 +162,19 @@ function capFirstLetterInName (fullName: string) {
...
-      <tr v-if="editingIndex != index && item.ui.showCeaseDateInput" data-cy="summary-table-row-cease">
-        <td colspan="5">
-          <IndividualPersonSummaryTableCessationDateEntry :individual="item" :index="index" />
-        </td>
-      </tr>
-
+      <IndividualPersonSummaryTableRowUpdating
+        v-if="item.ui.updating && editingIndex != index"
+        :si="item"
+      />
```
Comment (kialj876) [inline btr-web/btr-main-app/components/individual-person/summary-table/Index.vue:168]: this puts loader animation over the problem areas of the row that nuxt/ui renders incorrectly after we change them (does not pick up changes to the si field arrays properly) -- previously was handling this with setting the whole list of SIs to empty and then setting them back causing a flicker (was not pretty)

File: btr-web/btr-main-app/components/individual-person/summary-table/Index.vue [hunk 3]
```diff
@@ -193,77 +162,19 @@ function capFirstLetterInName (fullName: string) {
...
+      <IndividualPersonSummaryTableRow
+        v-else-if="editingIndex != index"
+        :si="item"
+        :index="index"
+        :action-btn="edit ? getActionButton(item, index) : undefined"
+        :action-dropdown-btns="getActionDropDownItems(item, index)"
+      />
```
Comment (kialj876) [inline btr-web/btr-main-app/components/individual-person/summary-table/Index.vue:175]: The deleted code from this section is all inside this component now. Allows for clearer logic in our v-ifs here

File: btr-web/btr-main-app/components/individual-person/summary-table/row/CessationEntry.vue [hunk 4]
```diff
@@ -0,0 +1,60 @@
+<script setup lang="ts">
```
Comment (kialj876) [inline btr-web/btr-main-app/components/individual-person/summary-table/row/CessationEntry.vue:1]: this component was just moved/renamed to be inside '/row'. Not sure why git didn't pick up on it just being a moved file (code is unchanged here)

File: btr-web/btr-main-app/stores/significant-individuals.ts [hunk 5]
```diff
@@ -86,19 +108,16 @@ export const useSignificantIndividuals = defineStore('significantIndividuals', (
...
   const filingRemoveSI = (index: number) => {
     allEditableSIs.value.splice(index, 1)
+    _applyUpdatingSI()
   }
 
   /** Remove the significant individual at the given index */
   const undoSIChanges = (index: number) => {
     const origIndex = allEditableSIs.value[index].ui.origIndex
-    if (origIndex) {
+    if (origIndex !== undefined) {
```
Comment (kialj876) [inline btr-web/btr-main-app/stores/significant-individuals.ts:117]: bug fix for not being able to undo first item in si list

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-btr-dev--pr-190-bj7omiyq.web.app

File: btr-web/btr-main-app/components/individual-person/summary-table/row/Index.vue [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+      <p v-for="date in si.effectiveDates" :key="date.startDate">
+        {{
+          $t('texts.dateRange', {
+            start: date.startDate ? date.startDate : $t('labels.unknown'),
+            end: date.endDate ? date.endDate : $t('labels.current')
+          })
+        }}
+      </p>
+    </td>
+    <!-- NOTE: opacityClass is not applied to the edit column on purpose -->
```
Comment (hfekete) [inline btr-web/btr-main-app/components/individual-person/summary-table/row/Index.vue:92]: Was it not working applying opacityClass to tr and then just resetting opacity for this td ?

Comment (hfekete) [reply btr-web/btr-main-app/components/individual-person/summary-table/row/Index.vue:92]: Not a deal breaker, no need to update it right now for the sake of time.

**Review by hfekete — APPROVED** [review]
Thx for all little small bugfixes!

File: btr-web/btr-main-app/components/individual-person/summary-table/row/Index.vue (hunk 6 above)
Comment (kialj876) [reply btr-web/btr-main-app/components/individual-person/summary-table/row/Index.vue:92]: I did not know there was a way to do what you are suggesting. I thought all children of a parent element with a certain opacity were stuck with it (unless they want less opacity)

**Pull request merged**